x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/esql_autocomplete/use_esql_fields_options.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EuiComboBoxOptionOption } from '@elastic/eui';
+import type { ExpressionsStart, Datatable } from '@kbn/expressions-plugin/public';
+
+import { useQuery } from '@tanstack/react-query';
+
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+
+import { getEsqlQueryConfig } from '../../logic/get_esql_query_config';
+import type { FieldType } from '../../logic/esql_validator';
+
+export const esqlToOptions = (
+  data: { error: unknown } | Datatable | undefined,
+  fieldType?: FieldType
+) => {
+  if (data && 'error' in data) {
+    return [];
+  }
+
+  const options = (data?.columns ?? [])
+    .filter(({ meta }) => {
+      // if fieldType absent, we do not filter columns by type
+      if (!fieldType) {
+        return true;
+      }
+      return fieldType === meta.type;
+    })
+    .map(({ id }) => ({ label: id }));
+
+  return options;
+};
+
+type UseEsqlFieldOptions = (
+  esqlQuery: string | undefined,
+  fieldType: FieldType
+) => {
+  isLoading: boolean;
+  options: Array<EuiComboBoxOptionOption<string>>;
+};
+
+/**
+ * fetches ES|QL fields and convert them to Combobox options
+ */
+export const useEsqlFieldOptions: UseEsqlFieldOptions = (esqlQuery, fieldType) => {
+  const kibana = useKibana<{ expressions: ExpressionsStart }>();
```

Comment:
NIT: the Security `useKibana` hook exported from the `common/lib/kibana` is already typed

Reply:
does not have `expressions` defined, so had to extend it

```


proc [tsc] x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/esql_autocomplete/use_esql_fields_options.ts:53:11 - error TS2339: Property 'expressions' does not exist on type 'Partial<CoreStart>'.
--
  | 2023-09-28 08:57:26 UTC | proc [tsc]
  | 2023-09-28 08:57:26 UTC | proc [tsc] 53   const { expressions } = kibana.services;
  | 2023-09-28 08:57:26 UTC | proc [tsc]              ~~~~~~~~~~~
  | 2023-09-28 08:57:26 UTC | proc [tsc]
  | 2023-09-28 08:57:26 UTC | proc [tsc]
  | 2023-09-28 08:57:26 UTC | proc [tsc] Found 1 error.
  | 2023-09-28 08:57:26 UTC | proc [tsc]



````